### PR TITLE
[USER_OPS_INDEXER] support v0.7 EntryPoint

### PR DESCRIPTION
**Pull request by k1rill-fedoseev**
# Changelog
* Add parallel support [v0.7 entrypoint](https://github.com/eth-infinitism/account-abstraction/releases/tag/v0.7.0) indexer
* Refactor common indexer logic
* Add optional support for events http polling instead of requiring websockets pubsub connection

### CODE REVIEW

File: user-ops-indexer/user-ops-indexer-logic/src/indexer/base_indexer.rs [hunk 1]
```diff
@@ -0,0 +1,538 @@
...
+            .block_hash
+            .ok_or(anyhow::anyhow!("unexpected pending log"))?;
+        Ok(Self {
+            tx_hash,
+            block_hash,
+        })
+    }
+}
+
+pub trait IndexerLogic {
```
Comment (sevenzing) [inline user-ops-indexer/user-ops-indexer-logic/src/indexer/base_indexer.rs:54]: I would rename IndexerLogic to BaseIndexer or rename this file from `base_indexer.rs` to `indexer_logic.rs`

File: user-ops-indexer/user-ops-indexer-server/src/indexer.rs [hunk 2]
```diff
@@ -1,33 +1,59 @@
...
     db_connection: DatabaseConnection,
 ) -> Result<(), anyhow::Error> {
     tracing::info!("connecting to rpc");
 
-    let ws_client = Ws::connect_with_reconnects(settings.indexer.rpc_url, 20).await?;
-    let client = Provider::new(ws_client);
+    let db_connection = Arc::new(db_connection);
+
+    let transport = CommonTransport::new(settings.indexer.rpc_url.clone()).await?;
+    let supports_subscriptions = matches!(transport, CommonTransport::Ws(_));
```
Comment (sevenzing) [inline user-ops-indexer/user-ops-indexer-server/src/indexer.rs:16]: I guess it's better to move this logic of "does support subscription" to CommonTransport then

like `let supports_subscriptions = transport.support_subscription()`

File: user-ops-indexer/user-ops-indexer-logic/src/indexer/base_indexer.rs (hunk 1 above)
Comment (sevenzing) [reply user-ops-indexer/user-ops-indexer-logic/src/indexer/base_indexer.rs:54]: oh, we gave IndexerLogic and Indexer, nvm then

**Review by rimrakhimov — APPROVED** [review]

File: user-ops-indexer/user-ops-indexer-proto/proto/user-ops-indexer.proto [hunk 3]
```diff
@@ -165,6 +165,18 @@ message RawUserOp {
   string signature = 11;
 }
 
+message RawUserOpV07 {
+  string sender = 1;
+  string nonce = 2;
+  string init_code = 3;
+  string call_data = 4;
+  string accountGasLimits = 5; // call_gas_limit + verification_gas_limit
```
[the remark this answers is not in the document]
Comment (k1rill-fedoseev) [reply user-ops-indexer/user-ops-indexer-proto/proto/user-ops-indexer.proto]: Thanks, just a copy paste error

**Pull request merged**